go/vt/vtgate/safe_session.go

```diff
@@ -103,34 +136,43 @@ func (session *SafeSession) AutocommitApproval() bool {
 	defer session.mu.Unlock()
 
 	if session.autocommitState == autocommitted {
-		panic("BUG: AutocommitToken: unexpected autocommit state")
+		// Unreachable.
+		return false
 	}
 
-	if session.autocommitState == autocommittable && len(session.ShardSessions) == 0 {
+	if session.autocommitState == autocommittable {
```

Comment:
`autocommittable` is sometimes spelt with one `t` like in the `func SetAutoCommitable`. Do we want to change all instances to be consistent?

Reply:
I'll rename the function.